apps/performance-tests/src/hierarchies/StatelessHierarchyProvider.ts

```diff
@@ -24,15 +26,15 @@ import {
   ECSqlQueryExecutor,
   ECSqlQueryReaderOptions,
 } from "@itwin/presentation-shared";
-import { asyncScheduler, expand, filter, finalize, from, observeOn, of, tap } from "rxjs";
 import { LOGGER } from "../util/Logging";
 
 interface ProviderOptionsBase {
   rowLimit?: number | "unbounded";
-  getHierarchyFactory(imodelAccess: ECSchemaProvider & ECClassHierarchyInspector): HierarchyDefinition;
+  getHierarchyFactory(imodelAccess: ECSchemaProvider & ECClassHierarchyInspector, idsCache?: ModelsTreeIdsCache): HierarchyDefinition;
```

Comment:
I don't think we need to add `idsCache` in here - it can be passed to `getHierarchyFactory` through context. E.g. in `ModelsTree.test.ts`:

```ts
const provider = new StatelessHierarchyProvider({
  imodelAccess,
  getHierarchyFactory: (imodelAccess) => new ModelsTreeDefinition({ imodelAccess, idsCache }),
  filtering,
});
```

Reply:
Removed